radix-engine/src/engine/call_frame.rs

```diff
@@ -404,12 +411,13 @@ impl<'r, 'l, L: ReadableSubstateStore> CallFrame<'r, 'l, L> {
     pub fn invoke_snode(
         &mut self,
         snode_ref: SNodeRef,
+        method_name: &str,
         call_data: ScryptoValue,
     ) -> Result<ScryptoValue, RuntimeError> {
         let function = if let Value::Enum { name, .. } = &call_data.dom {
             name.clone()
         } else {
-            return Err(RuntimeError::InvalidInvocation);
+            method_name.to_string()
         };
```

Comment:
Why do we need this logic? I thought you were using `invoke_snode2` for the `name + input` style.

I guess I'm also asking what the strategy for this massive refactoring is.